app/models/advisory_lock.rb

```diff
@@ -0,0 +1,19 @@
+class AdvisoryLock
+  class << self
+    def allocate_sip_trunk_channel(sip_trunk, options = {}, &block)
+      with_lock("allocate_sip_trunk_channel:#{sip_trunk.id}", options, &block)
```

Comment:
Should we inline the block easier for readers without trying to understand passing those arguments?